backend-implicaction/src/main/java/com/dynonuggets/refonteimplicaction/service/JobPostingService.java

```diff
@@ -53,4 +55,19 @@ public void deleteJobPosting(Long jobPostingId) {
                 .orElseThrow(() -> new NotFoundException("Impossible de supprimer l'offre, " + jobPostingId + " n'existe pas."));
         jobPostingRepository.delete(jobPosting);
     }
+
+    @Transactional
+    public JobPostingDto archiveOrUnarchiveJobPosting(Long jobPostingId) {
+        JobPosting jobPosting = jobPostingRepository.findById(jobPostingId)
+                .orElseThrow(() -> new NotFoundException("Impossible de supprimer l'offre, " + jobPostingId + " n'existe pas."));
+        if (jobPosting.getStatus() == statusRepository.getUnarchivedStatus()) {
+            jobPosting.setStatus(statusRepository.getArchivedStatus());
+        } else {
+            jobPosting.setStatus(statusRepository.getUnarchivedStatus());
+        }
+        final JobPosting save = jobPostingRepository.save(jobPosting);
+        return jobPostingAdapter.toDto(save);
```

Comment:
l'idée c'est d'introduire dans le modèle JobPosting (et du coup dans le dto) un attribut booleen `archive` pour au final simplifier la méthode comme ceci : 
```suggestion
       jobPosting.isArchive(!jobPosting.isArchive())
        final JobPosting save = jobPostingRepository.save(jobPosting);
        return jobPostingAdapter.toDto(save);
```

Reply:
du coup ce serait un champ persisté en base ? ou un champ propre à l'entité ?

Reply:
et en ce qui concerne le champ status, on en fait quoi ?